kpet/data.py

```diff
@@ -213,6 +213,8 @@ def __init__(self, data):
                     match=Class(PositivePattern),
                     dont_match=Class(NegativePattern),
                     waived=Boolean(),
+                    role=String(),
+                    extra_params=String(),
```

Comment:
Even though `role` is Beaker-specific, we'll need it as is for now. However, we have a good opportunity to remove Beaker-dependence and obscurity with `extra_params`. Let's just call it `environment`, or `env_vars`, or something like that. It will be clear, to the point, and non-Beaker specific. Finally make it `Dict(String())` to represent variable names/values directly, instead of having to paste XML into YAML, where we don't have to. Make the default be an empty dictionary too to simplify templates.